test/Microsoft.ML.Tests/Scenarios/SentimentPredictionTests.cs

```diff
@@ -143,6 +143,273 @@ public void TrainAndPredictSentimentModelTest()
+            IEnumerable<SentimentPrediction> predictions = tt.PredictorModels.Predict(sentiments);
+            Assert.Equal(2, predictions.Count());
+            Assert.True(predictions.ElementAt(0).Sentiment.IsFalse);
+            Assert.True(predictions.ElementAt(1).Sentiment.IsTrue);
+
+            predictions = tt.PredictorModels.Predict(sentiments);
+            Assert.Equal(2, predictions.Count());
+            Assert.True(predictions.ElementAt(0).Sentiment.IsFalse);
+            Assert.True(predictions.ElementAt(1).Sentiment.IsTrue);
+        }
+
+        [Fact]
+        public void CrossValidateSentimentModelTest()
+        {
+            string dataPath = GetDataPath(SentimentDataPath);
+            var pipeline = new LearningPipeline();
+
+            pipeline.Add(new Data.TextLoader(dataPath)
+            {
+                Arguments = new TextLoaderArguments
+                {
+                    Separator = new[] { '\t' },
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoaderColumn()
+                        {
+                            Name = "Label",
+                            Source = new [] { new TextLoaderRange(0) },
+                            Type = Runtime.Data.DataKind.Num
+                        },
+
+                        new TextLoaderColumn()
+                        {
+                            Name = "SentimentText",
+                            Source = new [] { new TextLoaderRange(1) },
+                            Type = Runtime.Data.DataKind.Text
+                        }
+                    }
+                }
+            });
+
+            pipeline.Add(new TextFeaturizer("Features", "SentimentText")
+            {
+                KeepDiacritics = false,
+                KeepPunctuations = false,
+                TextCase = TextNormalizerTransformCaseNormalizationMode.Lower,
+                OutputTokens = true,
+                StopWordsRemover = new PredefinedStopWordsRemover(),
+                VectorNormalizer = TextTransformTextNormKind.L2,
+                CharFeatureExtractor = new NGramNgramExtractor() { NgramLength = 3, AllLengths = false },
+                WordFeatureExtractor = new NGramNgramExtractor() { NgramLength = 2, AllLengths = true }
+            });
+
+            pipeline.Add(new FastTreeBinaryClassifier() { NumLeaves = 5, NumTrees = 5, MinDocumentsInLeafs = 2 });
+            pipeline.Add(new PredictedLabelColumnOriginalValueConverter() { PredictedLabelColumn = "PredictedLabel" });
+
+            IEnumerable<SentimentData> sentiments = new[]
+            {
+                new SentimentData
+                {
+                    SentimentText = "Please refrain from adding nonsense to Wikipedia."
+                },
+                new SentimentData
+                {
+                    SentimentText = "He is a CHEATER, and the article should say that."
+                }
+            };
```

Comment:
is this a same code in all 3 tests? Can you refactor it?

Reply:
Its ok for tests. 

---
In reply to: [191030443](https://github.com/dotnet/machinelearning/pull/212#discussion_r191030443) [](ancestors = 191030443)